pkg/operator/certrotationcontroller/externalloadbalancer.go

```
@@ -16,6 +16,10 @@ func (c *CertRotationController) syncExternalLoadBalancerHostnames() error {
 		return err
 	}
 	hostname := infrastructureConfig.Status.APIServerURL
+	if len(hostname) == 0 {
+		klog.Warningf("Failed to set external loadbalancer: APIServerURL is not set")
+		return nil
```

ignoring the error will make it not to reconcile later